### PR TITLE
cmake cleanups and enhancements

### PR DESCRIPTION
**Pull request by ssiloti**

### CODE REVIEW

File: libtorrent-rasterbarConfig.cmake [hunk 1]
```diff
@@ -0,0 +1 @@
+include("${CMAKE_CURRENT_LIST_DIR}/libtorrent-rasterbarTargets.cmake")
```
Comment (arvidn) [inline libtorrent-rasterbarConfig.cmake]: where does this file come from? "libtorrent-rasterbarTargets.cmake"

**Comment by arvidn** [comment]
lgtm, but I'm not a cmake expert

File: libtorrent-rasterbarConfig.cmake (hunk 1 above)
Comment (ssiloti) [inline libtorrent-rasterbarConfig.cmake]: It's generated by cmake when you do a "make install" or equivalent. You can see the reference to it on line 382 in CMakeLists.txt

**Comment by arvidn** [comment]
do you think there's a chance cmake could replace autoconf and automake for building tarballs?
I believe that is the main purpose of the makefiles right now.

**Comment by ssiloti** [comment]
Cmake comes with a tool called Cpack which can generate binary and source archives, I'm not sure how much of an improvement it would be over autotools. On Linux autotools is still often the preferred build system. I think it's mostly due to familiarity. Distro maintainers all know how to use autotools and often have their own tooling built around it.

**Pull request merged**